### PR TITLE
Release blocks in OrdinalsGroupingOperator

### LINKED ISSUE #100141: ESQL: Update OrdinalsGroupingOperator to close inbound pages
HashGroupingOperator has been updated to close inbound pages. Ordinals needs to do similar.

### PR DESCRIPTION
**Pull request by dnhatn**
Closes #100141

### CODE REVIEW

File: x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/OrdinalsGroupingOperator.java [hunk 1]
```diff
@@ -123,9 +122,6 @@ public void addInput(Page page) {
         checkState(needsInput(), "Operator is already finishing");
         requireNonNull(page, "page is null");
         DocVector docVector = page.<DocBlock>getBlock(docChannel).asVector();
-        if (docVector.getPositionCount() == 0) {
-            return;
-        }
```
Comment (nik9000) [inline x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/OrdinalsGroupingOperator.java:128]: We don't send empty blocks into Operators at this point anyway.

**Review by nik9000 — APPROVED** [review]
It looks right to me. I think we should grow an `OrdinalsGroupingOperatorTests` at some point soon. It's much easier to debug problems building with it. And it's runs the CrankyBreaker which is much better at finding weird failures when building than I am by hand. I think you've covered them all with the `try` blocks you have, but it's hard for me to know without the test.

**Review by ChrisHegarty — APPROVED** [review]
LGTM

**Comment by dnhatn** [comment]
> I think we should grow an OrdinalsGroupingOperatorTests at some point soon

++. I will do this in a follow-up.

**Comment by dnhatn** [comment]
@nik9000 @ChrisHegarty Thanks for reviews.

**Pull request merged**